src/pybind/mgr/cephadm/module.py

```diff
@@ -1456,35 +1456,72 @@ def add_host(self, spec: HostSpec) -> str:
         return self._add_host(spec)
 
     @handle_orch_error
-    def remove_host(self, host):
-        # type: (str) -> str
+    def remove_host(self, host: str, force: bool = False, offline: bool = False) -> str:
         """
         Remove a host from orchestrator management.
 
         :param host: host name
+        :param force: bypass running daemons check
+        :param offline: remove offline host
         """
-        # Verify if it is possible to remove the host safely
-        daemons = self.cache.get_daemons_by_host(host)
-        if daemons:
-            self.log.warning(f"Blocked {host} removal. Daemons running: {daemons}")
 
-            daemons_table = ""
-            daemons_table += "{:<20} {:<15}\n".format("type", "id")
-            daemons_table += "{:<20} {:<15}\n".format("-" * 20, "-" * 15)
+        # check if host is offline
+        host_offline = host in self.offline_hosts or self.inventory._inventory[host].get(
+            'status', '').lower() == 'offline'
+
+        if host_offline and not offline:
+            return "{} is offline, please use --offline and --force to remove this host.".format(host)
```

Comment:
No one is going to read the documentation. Should we add `can potentially cause data loss` to the message?